### PR TITLE
fix(dav): don't crash subscription on invalid calendar object

### LINKED ISSUE #48518: [Bug]: Failure on subscribing to a public Google calendar
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When using the calendar app to add a public Google calendar i get errors. Furthermore only a small selection of the events are shown.

An other Google calender is working without problems.

### Steps to reproduce

Happening with every try with my specific calendar.

### Expected behavior

This calendar is working like the others.

### Nextcloud Server version

30

### Operating system

Other

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "password": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "30.0.0.14",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "skeletondirectory": "",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtpsecure": "ssl",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "default_language": "de",
        "default_locale": "de_DE",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "default_phone_region": "DE",
        "defaultapp": "calendar",
        "loglevel": 0,
        "maintenance": false,
        "theme": "",
        "htaccess.RewriteBase": "\/",
        "maintenance_window_start": "3",
        "overwriteprotocol": "https",
        "app_install_overwrite": [
            "polls",
            "forms"
        ]
    }
}
```

### List of activated Apps

```shell
Enabled:
  - bruteforcesettings: 3.0.0
  - calendar: 5.0.0
  - calendar_resource_management: 0.8.0
  - cloud_federation_api: 1.13.0
  - contacts: 6.1.0
  - contactsinteraction: 1.11.0
  - dav: 1.31.1
  - federatedfilesharing: 1.20.0
  - files: 2.2.0
  - files_downloadlimit: 3.0.0
  - files_external: 1.22.0
  - files_pdfviewer: 3.0.0
  - files_reminders: 1.3.0
  - files_sharing: 1.22.0
  - files_trashbin: 1.20.1
  - files_versions: 1.23.0
  - forms: 4.2.4
  - group_everyone: 0.1.16
  - impersonate: 1.17.0
  - logreader: 3.0.0
  - lookup_server_connector: 1.18.0
  - nextcloud_announcements: 2.0.0
  - notifications: 3.0.0
  - oauth2: 1.18.1
  - oidc: 0.9.4
  - password_policy: 2.0.0
  - polls: 7.2.3
  - previewgenerator: 5.6.0
  - privacy: 2.0.0
  - provisioning_api: 1.20.0
  - serverinfo: 2.0.0
  - settings: 1.13.0
  - sharebymail: 1.20.0
  - text: 4.1.0
  - theming: 2.5.0
  - twofactor_backupcodes: 1.19.0
  - updatenotification: 1.20.0
  - viewer: 3.0.0
  - webapppassword: 24.9.0
  - workflowengine: 2.12.0
Disabled:
  - activity: 3.0.0 (installed 2.18.0)
  - admin_audit: 1.20.0
  - calendar.bak: 5.0.0
  - circles: 30.0.0-dev (installed 26.0.0)
  - comments: 1.20.1 (installed 1.16.0)
  - dashboard: 7.10.0 (installed 7.6.0)
  - encryption: 2.18.0
  - federation: 1.20.0 (installed 1.17.0)
  - firstrunwizard: 3.0.0 (installed 2.15.0)
  - phonetrack: 0.8.1 (installed 0.8.1)
  - photos: 3.0.2 (installed 2.2.0)
  - recommendations: 3.0.0 (installed 1.5.0)
  - related_resources: 1.5.0 (installed 1.1.0-alpha1)
  - support: 2.0.0 (installed 1.9.0)
  - survey_client: 2.0.0 (installed 1.14.0)
  - suspicious_login: 8.0.0 (installed 8.0.0)
  - systemtags: 1.20.0 (installed 1.16.0)
  - twofactor_nextcloud_notification: 4.0.0
  - twofactor_totp: 12.0.0-dev
  - user_ldap: 1.21.0
  - user_status: 1.10.0 (installed 1.6.0)
  - weather_status: 1.10.0 (installed 1.6.0)
  - webhook_listeners: 1.1.0-dev (installed 1.1.0-dev)
```

### Nextcloud Signing status

_No response_

### Nextcloud Logs

```shell
{
  "reqId": "kbmyKr2K2FEYkV3xFQex",
  "level": 0,
  "time": "2024-09-20T05:02:28+00:00",
  "remoteAddr": "10.89.0.133",
  "user": "...",
  "app": "no app in context",
  "method": "MKCOL",
  "url": "/remote.php/dav/calendars/.../...",
  "message": "dirty table reads: SELECT `displayname`, `refreshrate`, `calendarorder`, `calendarcolor`, `striptodos`, `stripalarms`, `stripattachments`, `id`, `uri`, `source`, `principaluri`, `lastmodified`, `synctoken` FROM `*PREFIX*calendarsubscriptions` WHERE `principaluri` = :dcValue1 ORDER BY `calendarorder` asc",
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36 Edg/128.0.0.0",
  "version": "30.0.0.14",
  "exception": {
    "Exception": "Exception",
    "Message": "dirty table reads: SELECT `displayname`, `refreshrate`, `calendarorder`, `calendarcolor`, `striptodos`, `stripalarms`, `stripattachments`, `id`, `uri`, `source`, `principaluri`, `lastmodified`, `synctoken` FROM `*PREFIX*calendarsubscriptions` WHERE `principaluri` = :dcValue1 ORDER BY `calendarorder` asc",
    "Code": 0,
    "Trace": [
      {
        "file": "/var/www/html/lib/private/DB/ConnectionAdapter.php",
        "line": 50,
        "function": "executeQuery",
        "class": "OC\\DB\\Connection",
        "type": "->",
        "args": [
          "SELECT `displayname`, `refreshrate`, `calendarorder`, `calendarcolor`, `striptodos`, `stripalarms`, `stripattachments`, `id`, `uri`, `source`, `principaluri`, `lastmodified`, `synctoken` FROM `*PREFIX*calendarsubscriptions` WHERE `principaluri` = :dcValue1 ORDER BY `calendarorder` asc",
          ["*** sensitive parameters replaced ***"],
          [2]
        ]
      },
      {
        "file": "/var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php",
        "line": 272,
        "function": "executeQuery",
        "class": "OC\\DB\\ConnectionAdapter",
        "type": "->",
        "args": [
          "SELECT `displayname`, `refreshrate`, `calendarorder`, `calendarcolor`, `striptodos`, `stripalarms`, `stripattachments`, `id`, `uri`, `source`, `principaluri`, `lastmodified`, `synctoken` FROM `*PREFIX*calendarsubscriptions` WHERE `principaluri` = :dcValue1 ORDER BY `calendarorder` asc",
          ["*** sensitive parameters replaced ***"],
          [2]
        ]
      },
      {
        "file": "/var/www/html/apps/dav/lib/CalDAV/CalDavBackend.php",
        "line": 2551,
        "function": "executeQuery",
        "class": "OC\\DB\\QueryBuilder\\QueryBuilder",
        "type": "->",
        "args": []
      },
      {
        "file": "/var/www/html/apps/dav/lib/CalDAV/WebcalCaching/RefreshWebcalService.php",
        "line": 162,
        "function": "getSubscriptionsForUser",
        "class": "OCA\\DAV\\CalDAV\\CalDavBackend",
        "type": "->",
        "args": ["*** sensitive parameters replaced ***"]
      },
      {
        "file": "/var/www/html/apps/dav/lib/CalDAV/WebcalCaching/RefreshWebcalService.php",
        "line": 40,
        "function": "getSubscription",
        "class": "OCA\\DAV\\CalDAV\\WebcalCaching\\RefreshWebcalService",
        "type": "->",
        "args": [
          "*** sensitive parameters replaced ***",
          "*** sensitive parameters replaced ***"
        ]
      },
      {
        "file": "/var/www/html/apps/dav/lib/Listener/SubscriptionListener.php",
        "line": 45,
        "function": "refreshSubscription",
        "class": "OCA\\DAV\\CalDAV\\WebcalCaching\\RefreshWebcalService",
        "type": "->",
        "args": [
          "*** sensitive parameters replaced ***",
          "*** sensitive parameters replaced ***"
        ]
      },
      {
        "file": "/var/www/html/lib/private/EventDispatcher/ServiceEventListener.php",
        "line": 68,
        "function": "handle",
        "class": "OCA\\DAV\\Listener\\SubscriptionListener",
        "type": "->",
        "args": [["OCA\\DAV\\Events\\SubscriptionCreatedEvent"]]
      },
      {
        "file": "/var/www/html/3rdparty/symfony/event-dispatcher/EventDispatcher.php",
        "line": 230,
        "function": "__invoke",
        "class": "OC\\EventDispatcher\\ServiceEventListener",
        "type": "->",
        "args": [
          ["OCA\\DAV\\Events\\SubscriptionCreatedEvent"],
          "OCA\\DAV\\Events\\SubscriptionCreatedEvent",
          ["Symfony\\Component\\EventDispatcher\\EventDispatcher"]
        ]
      },
      {
        "file": "/var/www/html/3rdparty/symfony/event-dispatcher/EventDispatcher.php",
        "line": 59,
        "function": "callListeners",
        "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
        "type": "->",
        "args": [
          [["Closure"]],
          "OCA\\DAV\\Events\\SubscriptionCreatedEvent",
          ["OCA\\DAV\\Events\\SubscriptionCreatedEvent"]
        ]
      },
      {
        "file": "/var/www/html/lib/private/EventDispatcher/EventDispatcher.php",
        "line": 67,
        "function": "dispatch",
        "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
        "type": "->",
        "args": [
          ["OCA\\DAV\\Events\\SubscriptionCreatedEvent"],
          "OCA\\DAV\\Events\\SubscriptionCreatedEvent"
        ]
      },
      {
        "file": "/var/www/html/lib/private/EventDispatcher/EventDispatcher.php",
        "line": 79,
        "function": "dispatch",
        "class": "OC\\EventDispatcher\\EventDispatcher",
        "type": "->",
        "args": [
          "OCA\\DAV\\Events\\SubscriptionCreatedEvent",
          ["OCA\\DAV\\Events\\SubscriptionCreatedEvent"]
        ]
      },
      {
        "file": "/var/www/html/apps/dav/lib/CalDAV/CalDavBackend.php",
        "line": 2622,
        "function": "dispatchTyped",
        "class": "OC\\EventDispatcher\\EventDispatcher",
        "type": "->",
        "args": [["OCA\\DAV\\Events\\SubscriptionCreatedEvent"]]
      },
      {
        "file": "/var/www/html/3rdparty/sabre/dav/lib/CalDAV/CalendarHome.php",
        "line": 249,
        "function": "createSubscription",
        "class": "OCA\\DAV\\CalDAV\\CalDavBackend",
        "type": "->",
        "args": ["*** sensitive parameters replaced ***"]
      },
      {
        "file": "/var/www/html/apps/dav/lib/CalDAV/CalendarHome.php",
        "line": 77,
        "function": "createExtendedCollection",
        "class": "Sabre\\CalDAV\\CalendarHome",
        "type": "->",
        "args": ["*** sensitive parameters replaced ***"]
      },
      {
        "file": "/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 1199,
        "function": "createExtendedCollection",
        "class": "OCA\\DAV\\CalDAV\\CalendarHome",
        "type": "->",
        "args": ["*** sensitive parameters replaced ***"]
      },
      {
        "file": "/var/www/html/3rdparty/sabre/dav/lib/DAV/CorePlugin.php",
        "line": 560,
        "function": "createCollection",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": ["*** sensitive parameters replaced ***"]
      },
      {
        "file": "/var/www/html/3rdparty/sabre/event/lib/WildcardEmitterTrait.php",
        "line": 89,
        "function": "httpMkcol",
        "class": "Sabre\\DAV\\CorePlugin",
        "type": "->",
        "args": [["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"]]
      },
      {
        "file": "/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 472,
        "function": "emit",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": [
          "method:MKCOL",
          [["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"]]
        ]
      },
      {
        "file": "/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 253,
        "function": "invokeMethod",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": [["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"]]
      },
      {
        "file": "/var/www/html/apps/dav/lib/Server.php",
        "line": 370,
        "function": "start",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": []
      },
      {
        "file": "/var/www/html/apps/dav/appinfo/v2/remote.php",
        "line": 19,
        "function": "exec",
        "class": "OCA\\DAV\\Server",
        "type": "->",
        "args": []
      },
      {
        "file": "/var/www/html/remote.php",
        "line": 146,
        "args": ["/var/www/html/apps/dav/appinfo/v2/remote.php"],
        "function": "require_once"
      }
    ],
    "File": "/var/www/html/lib/private/DB/Connection.php",
    "Line": 402,
    "message": "dirty table reads: SELECT `displayname`, `refreshrate`, `calendarorder`, `calendarcolor`, `striptodos`, `stripalarms`, `stripattachments`, `id`, `uri`, `source`, `principaluri`, `lastmodified`, `synctoken` FROM `*PREFIX*calendarsubscriptions` WHERE `principaluri` = :dcValue1 ORDER BY `calendarorder` asc",
    "tables": ["oc_calendarsubscriptions"],
    "reads": ["oc_calendarsubscriptions"],
    "exception": [],
    "CustomMessage": "dirty table reads: SELECT `displayname`, `refreshrate`, `calendarorder`, `calendarcolor`, `striptodos`, `stripalarms`, `stripattachments`, `id`, `uri`, `source`, `principaluri`, `lastmodified`, `synctoken` FROM `*PREFIX*calendarsubscriptions` WHERE `principaluri` = :dcValue1 ORDER BY `calendarorder` asc"
  },
  "id": "66ed01eb73efe"
}
```

### Additional info

image

### PR DESCRIPTION
**Pull request by escoand**
* Resolves: https://github.com/nextcloud/server/issues/48518

## Summary

I had two different problems here when subscribing to an external calendar. 
When importing the same calendar these wrong objects get skipped.
But subscribing crashes at this point completely causing a partially imported calendar.

Nextcloud 29.0.7 is not affected.

```json
{
  "reqId": "3KSLP9rLSEFtuslDIPBy",
  "level": 3,
  "time": "2024-09-25T10:04:22+00:00",
  "remoteAddr": "...",
  "user": "...",
  "app": "webdav",
  "method": "MKCOL",
  "url": "/remote.php/dav/calendars/...",
  "message": "The supplied iCalendar datetime value is incorrect: 20181202 00:00:00",
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36 Edg/129.0.0.0",
  "version": "30.0.0.14",
  "exception": {
    "Exception": "Sabre\\VObject\\InvalidDataException",
    "Message": "The supplied iCalendar datetime value is incorrect: 20181202 00:00:00",
    "Code": 0,
    "Trace": /*...*/,
    "File": "/var/www/html/3rdparty/sabre/vobject/lib/DateTimeParser.php",
    "Line": 40,
    "message": "The supplied iCalendar datetime value is incorrect: 20181202 00:00:00",
    "exception": {},
    "CustomMessage": "The supplied iCalendar datetime value is incorrect: 20181202 00:00:00"
  }
}
```

and

```json
{
  "reqId": "s9qaIfKbFDRspoMdIoXP",
  "level": 0,
  "time": "2024-09-25T09:56:50+00:00",
  "remoteAddr": "...",
  "user": "...",
  "app": "webdav",
  "method": "MKCOL",
  "url": "/remote.php/dav/calendars/...",
  "message": "This recurrence rule does not generate any valid instances",
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36 Edg/129.0.0.0",
  "version": "30.0.0.14",
  "exception": {
    "Exception": "Sabre\\DAV\\Exception\\Forbidden",
    "Message": "This recurrence rule does not generate any valid instances",
    "Code": 0,
    "Trace": /*...*/,
    "File": "/var/www/html/apps/dav/lib/CalDAV/CalDavBackend.php",
    "Line": 3022,
    "message": "This recurrence rule does not generate any valid instances",
    "exception": {},
    "CustomMessage": "This recurrence rule does not generate any valid instances"
  }
}
```

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/dav/lib/CalDAV/WebcalCaching/RefreshWebcalService.php [hunk 1]
```diff
@@ -101,7 +102,13 @@ public function refreshSubscription(string $principalUri, string $uri) {
 					continue;
 				}
 
-				$denormalized = $this->calDavBackend->getDenormalizedData($vObject->serialize());
+				try {
+					$denormalized = $this->calDavBackend->getDenormalizedData($vObject->serialize());
+				} catch (InvalidDataException|Forbidden $ex) {
+					$this->logger->error('Unable to denormalize calendar object from subscription {subscriptionId}', ['exception' => $ex, 'subscriptionId' => $subscription['id'], 'source' => $subscription['source']]);
```
Comment (kesselb) [inline apps/dav/lib/CalDAV/WebcalCaching/RefreshWebcalService.php:108]: I would suggest warning here, error seems a bit high.

Comment (escoand) [reply apps/dav/lib/CalDAV/WebcalCaching/RefreshWebcalService.php:108]: This was copied from 30 lines below. Isn't this the same severity?

Comment (kesselb) [reply apps/dav/lib/CalDAV/WebcalCaching/RefreshWebcalService.php:108]: 30 lines below should also be a warning then ;)

Comment (SebastianKrupinski) [reply apps/dav/lib/CalDAV/WebcalCaching/RefreshWebcalService.php:108]: Hey, I agree with @kesselb this should just be a warning.

Errors have the side effect of showing up on the Administration -> Overview page, which will make admins think there is something wrong when its just a bad subscription. 

As for the $logger->error 30 lines lower, you're more than welcome to change that to a warning, a warning makes more sense

Comment (joshtrichards) [reply apps/dav/lib/CalDAV/WebcalCaching/RefreshWebcalService.php]: Too bad we can't only notify the relevant end-user...

Comment (SebastianKrupinski) [reply apps/dav/lib/CalDAV/WebcalCaching/RefreshWebcalService.php]: @joshtrichards that would be a good feature, using the notification module to notify the user instead would be great.

**Review by SebastianKrupinski — APPROVED** [review]
Looks good.

@escoand only one issue you need to sign the PR.

File: apps/dav/lib/CalDAV/WebcalCaching/RefreshWebcalService.php (hunk 1 above)
Comment (escoand) [reply apps/dav/lib/CalDAV/WebcalCaching/RefreshWebcalService.php]: This code is also called on updates later, if I am right. So wouldn't this create a notification every time the ical gets updated?

**Comment by escoand** [comment]
@SebastianKrupinski done.

**Comment by kesselb** [comment]
To fix the failing tests:

```
Index: apps/dav/tests/unit/CalDAV/WebcalCaching/RefreshWebcalServiceTest.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/apps/dav/tests/unit/CalDAV/WebcalCaching/RefreshWebcalServiceTest.php b/apps/dav/tests/unit/CalDAV/WebcalCaching/RefreshWebcalServiceTest.php
--- a/apps/dav/tests/unit/CalDAV/WebcalCaching/RefreshWebcalServiceTest.php	(revision 6be00432b75a80a246246883c5fa955ce803f3d8)
+++ b/apps/dav/tests/unit/CalDAV/WebcalCaching/RefreshWebcalServiceTest.php	(date 1727814673893)
@@ -302,7 +302,7 @@
 			->willThrowException($badRequestException);
 
 		$this->logger->expects(self::once())
-			->method('error')
+			->method('warning')
 			->with('Unable to create calendar object from subscription {subscriptionId}', ['exception' => $badRequestException, 'subscriptionId' => '42', 'source' => 'webcal://foo.bar/bla2']);
 
 		$refreshWebcalService->refreshSubscription('principals/users/testuser', 'sub123');

```

**Comment by SebastianKrupinski** [comment]
@escoand... @kesselb is correct due to the change from a error to a warning our ci unit tests are failing...

```
1) OCA\DAV\Tests\unit\CalDAV\WebcalCaching\RefreshWebcalServiceTest::testRunCreateCalendarNoException with data set #0 ('BEGIN:VCALENDAR\r\nVERSION:2....AR\r\n', 'text/calendar;charset=utf8', 'BEGIN:VCALENDAR\r\nVERSION:2....AR\r\n')
Expectation failed for method name is "error" when invoked 1 time(s).
Method was expected to be called 1 times, actually called 0 times.

4) OCA\DAV\Tests\unit\CalDAV\WebcalCaching\RefreshWebcalServiceTest::testRunCreateCalendarBadRequest with data set #0 ('BEGIN:VCALENDAR\r\nVERSION:2....AR\r\n', 'text/calendar;charset=utf8', 'BEGIN:VCALENDAR\r\nVERSION:2....AR\r\n')
Expectation failed for method name is "error" when invoked 1 time(s).
Method was expected to be called 1 times, actually called 0 times.

```

Do you mind adjusting them? The file is "apps/dav/tests/unit/CalDAV/WebcalCaching/RefreshWebcalServiceTest.php" line 255 and 305. Both lines should be "->method('warning')"

Thanks! I would do it for you but its a bit of a pain because you are on a forked patch.

**Comment by kesselb** [comment]
Thanks again. 

Cherry-picked your commits into https://github.com/nextcloud/server/pull/48519.

A part of our CI is skipped for forks because they don't work currently, but some of the skipped workflows are required to merge a pr :see_no_evil:

**Pull request closed without merging**